app/assets/stylesheets/darkswarm/modals.css.scss

```diff
@@ -28,7 +28,7 @@ dialog, .reveal-modal {
   // Medium and up - when modal IS NOT full screen
   @media only screen and (min-width: 641px) {
     top: 10%;
-    max-height: 120%;
+    max-height: 100%;
```

Comment:
Can you explain why you are doing this change? And does it open #1766 fixed in #2055 again?

It's good practice to explain the why in the commit message.

Reply:
oh, sorry, I forgot to explain this one!

I changed it to 100% so that the modal is never outside the viewport. Allowing a height over 100% makes content go outside the viewport without scrollbars...

Thanks for the reference to #2055! I have just tested the registration process with this new value and it looks good according to my tests.

@HugsDaniel does this make sense to you? I see your comment here: https://github.com/openfoodfoundation/openfoodnetwork/pull/2055#issuecomment-366676755
Why would you ever let max-height be bigger than 100%? That means you remove scroll bars and the content will not show. I think 100% is always the correct value, so that if the content doesn't fit, you get the scroll bars. That's how all modals should work. Is there an example where you think this 100% value will not work?
Thanks!

Reply:
I see. Well, I have retested the registration process and it looks good with max-height 100%: it makes the modal scroll, not the screen. I think this is fine. But lets wait for someone else's feedback.
Thanks!

Reply:
We'll see when we test but my experience of cookies banner is that you can scroll the page behind it but it doesn't scroll the modal. But for the registration process on modal you will definitely need to scroll as some modal pages have too many info to fit on small mobile screens... anyway I don't get the technical side so happy to see when we test what it looks like :-)

Reply:
Referencing PR #2544.

In summary, I discovered that there were issues with the JS modal library (`angular-bootstrap`) and the styles we are using:

1. The 10px space at the top is actually intended to be 10% in the styles. The JS library is just treating it as px, I don't know if by design.
2. There is a race condition between the -25% animation of the modal and `focus()` in the JS library which causes the page to scroll up (if still possible).
3. Offset from the top plus the height of the modal (on any screen size) should not exceed 100%, or else some content could be hidden. The change of @luisramos0 decreasing `max-height` for large screens from 120% to 100% is a big improvement. #2544 also addresses the lack of `max-height` for small screens.

I see that this is already merged. Maybe we can move discussion of the modal dimensions to #2544? I've just rebased it upon the updated master. I'll see how it affects the cookies policy modal, and also check #1766.